src/test/custom_node_output_allocator_test.cpp

```diff
@@ -0,0 +1,103 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../custom_node_output_allocator.hpp"
+
+using namespace ovms;
+
+class NodeLibraryTest : public NodeLibrary {
+    static bool releaseBufferCalled;
+
+public:
+    ~NodeLibraryTest();
+    static int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength);
+    static int releaseBuffer(struct CustomNodeTensor* output);
+    static int releaseTensors(struct CustomNodeTensor* outputs);
+};
+
+NodeLibraryTest::~NodeLibraryTest() {
```

Comment:
Minor:
I would rename the class name to something different since *Test suffix is usually for the Gtest fixtures. 

Maybe NodeLibraryCheckingReleaseCalled -> then it  would be clear just from the name of the class what it tries to achieve.